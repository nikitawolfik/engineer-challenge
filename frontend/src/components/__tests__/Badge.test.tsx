/**
 * @jest-environment jsdom
 */

import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Badge from "components/Badge";

describe("Search input", () => {
  test("Snapshot", async () => {
    const { container, rerender } = render(<Badge status="ACTIVE" />);

    expect(container).toMatchSnapshot();

    rerender(<Badge status="CANCELLED" />);
    expect(container).toMatchSnapshot();

    rerender(<Badge status="DROPPED_OUT" />);
    expect(container).toMatchSnapshot();

    rerender(<Badge status="PENDING" />);
    expect(container).toMatchSnapshot();
  });
});
