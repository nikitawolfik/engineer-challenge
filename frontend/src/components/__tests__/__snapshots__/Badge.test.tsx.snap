// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search input Snapshot 1`] = `
<div>
  <p
    class="inline-block rounded-full	py-1 px-4 font-semibold text-xs tc-green-100 bg-green-100"
  >
    ACTIVE
  </p>
</div>
`;

exports[`Search input Snapshot 2`] = `
<div>
  <p
    class="inline-block rounded-full	py-1 px-4 font-semibold text-xs tc-red-100 bg-red-100"
  >
    CANCELLED
  </p>
</div>
`;

exports[`Search input Snapshot 3`] = `
<div>
  <p
    class="inline-block rounded-full	py-1 px-4 font-semibold text-xs tc-red-100 bg-red-100"
  >
    DROPPED_OUT
  </p>
</div>
`;

exports[`Search input Snapshot 4`] = `
<div>
  <p
    class="inline-block rounded-full	py-1 px-4 font-semibold text-xs tc-yellow-600 bg-yellow-100"
  >
    PENDING
  </p>
</div>
`;
